src/lib/storage/value_segment/value_segment_iterable.hpp

```diff
@@ -129,12 +144,16 @@ class ValueSegmentIterable : public PointAccessibleSegmentIterable<ValueSegmentI
 
     std::ptrdiff_t distance_to(const Iterator& other) const { return other._value_it - _value_it; }
 
-    SegmentPosition<T> dereference() const { return SegmentPosition<T>{*_value_it, *_null_value_it, _chunk_offset}; }
+    SegmentPosition<T> dereference() const {
+//      _segment->access_statistics().on_iterator_dereference(1);
```

Comment:
I guess this PR is just a snapshot and you deactivated this for testing reasons, correct?

Reply:
It will be removed beacuse it is to expensive.